server/game/GameActions/DuelAction.ts

```diff
@@ -102,9 +103,17 @@ export class DuelAction extends CardGameAction {
             context.game.addMessage('The duel cannot proceed as at least one participant for each side has to be in play');
             return;
         }
+        var duel = new Duel(context.game, properties.challenger, cards, properties.type, properties.statistic);
+        if(properties.duringDuelAction) {
+            properties.duringDuelAction.duration = Durations.UntilEndOfDuel;
+            context.duel = duel;
```

Comment:
Ok, I think I get what you're trying to do, but I don't think it will work - does this fix pass tests?  The context of the ability which initiated the duel is not the same as the context of the lasting effect once it's created.  I can see an argument that it perhaps should be, but I think that will require a fairly major rewrite to the effect engine.